akka-remote/src/main/scala/akka/remote/artery/LruBoundedCache.scala

```diff
@@ -0,0 +1,234 @@
+      val otherProbeDistance = probeDistanceOf(position)
+      if (values(position) eq null) {
+        None
+      } else if (probeDistance > otherProbeDistance) {
+        None
+      } else if (hashes(position) == h && k == keys(position)) {
+        Some(values(position))
+      } else {
+        findOrCalculate((position + 1) & Mask, probeDistance + 1)
+      }
+    }
+
+    findOrCalculate(position = h & Mask, probeDistance = 0)
+  }
+
+  final def getOrCompute(k: K): V = {
+    val h = hash(k)
+    epoch += 1
+
+    @tailrec def findOrCalculate(position: Int, probeDistance: Int): V = {
+      if (values(position) eq null) {
+        val value = calculate(k)
+        keys(position) = k
+        values(position) = value
+        hashes(position) = h
+        epochs(position) = epoch
+        value
+      } else {
+        val otherProbeDistance = probeDistanceOf(position)
+        // If probe distance of the element we try to get is larger than the current slot's, then the element cannot be in
+        // the table since because of the Robin-Hood property we would have swapped it with the current element.
+        if (probeDistance > otherProbeDistance) {
+          val otherEpoch = epochs(position)
+
+          // Check whether colliding entry (where we would like to insert) is too old so we can safely overwrite
+          if (epoch - otherEpoch < evictAgeThreshold) {
+            val value = calculate(k)
+            // Existing entry is not too old, but we can take its slot and kick it out
+            val otherKey = keys(position)
+            val otherValue = values(position)
+            val otherHash = hashes(position)
+
+            keys(position) = k
+            values(position) = value
+            hashes(position) = h
+            epochs(position) = epoch
+
+            // Move the entry that we kicked out to somewhere else
+            move((position + 1) & Mask, otherKey, otherHash, otherValue, otherEpoch, otherProbeDistance + 1)
+            // Then we are done
+            value
+          } else {
+            // Remove old entry properly (rebalance) before attempting to enter new value
+            removeAt(position)
+            // Then try again
+            findOrCalculate(position, probeDistance)
+          }
+        } else if (hashes(position) == h && k == keys(position)) {
+          // Update usage
+          epochs(position) = epoch
+          values(position)
+        } else {
+          // This is not our slot yet
+          findOrCalculate((position + 1) & Mask, probeDistance + 1)
+        }
+      }
+    }
+
+    findOrCalculate(position = h & Mask, probeDistance = 0)
+  }
+
+  @tailrec private def removeAt(position: Int): Unit = {
+    val next = (position + 1) & Mask
+    if ((values(next) eq null) || probeDistanceOf(next) == 0) {
+      // Next is not movable, just empty this slot
+      values(position) = null.asInstanceOf[V]
```

Comment:
The compiler says so :( Which is weird since `V <: AnyRef` but who am I to question the compiler :D